### DISCUSSION TITLE
REST API: Why do we need to set the `status` field to create a new branch?

### DISCUSSION THREAD

**Discussion by LucaNicosia** [netboxlabs/netbox-branching#173] [Q&A] [answered]
I'm trying to manager netbox branches using the REST APIs, but when I try to create one (following the example [here](https://github.com/netboxlabs/netbox-branching/blob/develop/docs/rest-api.md#creating-a-branch)), I get the following error.
``` json
{
    "status": [
        "This field is required."
    ]
}
```

If I instead set this as the body of the request, the branch is created
``` json
{
    "name": "my-new-branch",
    "status": "new"
}
```

Since I can set the status, I can also do the following:
``` json
{
    "name": "my-new-branch",
    "status": "ready"
}
```
which doesn't make sense to me

My question is: there a reason why we would need to set the `status` to `ready` (or any other status except `new`) when creating a new branch? If not, wouldn't be better to force the status to be `new` when creating a new branch?

**Comment by deliciouslytyped** [comment]
I have this same issue.

**Comment by jeremystretch** [comment]
Please submit a bug report for this.

**Comment by jchambers2012** [comment, marked as answer]
I also have this bug - submitted #225

**Comment by mrmrcoleman** [comment]
Closing this as https://github.com/netboxlabs/netbox-branching/issues/225 is now closed
